src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -208,13 +208,13 @@ static ulong SoftwareFallback(ulong a, ulong b, out ulong low)
                 uint bl = (uint)b;
                 uint bh = (uint)(b >> 32);
 
-                ulong mull = ((ulong)al) * bl;
-                ulong t = ((ulong)ah) * bl + (mull >> 32);
-                ulong tl = ((ulong)al) * bh + (uint)t;
+                ulong mull = BigMul(al, bl);
```

Comment:
This is de-optimization. The internal `BigMul(uint,uint)` method produces poor code. Check the discussion at https://github.com/dotnet/runtime/pull/100375#discussion_r1544048252

Reply:
I reversed this part. Just like I wrote in summary and @MichalPetryka suggested in the thread.
Should we not remove the BMI2 path that causes poor codegen?